src/app/common-components/header/header.component.ts

```diff
@@ -10,15 +9,16 @@ import { User } from "src/app/login/_models/user";
 })
 export class HeaderComponent implements OnInit {
     isAuthenticated = false;
+    userName: string;
     constructor(private authService: AuthService, private router: Router) {}
 
     ngOnInit() {
-        this.authService.isAuthenticated$.subscribe(isAuthenticated => (this.isAuthenticated = isAuthenticated));
-    }
-
-    getUserInfo() {
-        const user: User = this.authService.getUserInfo();
-        console.log("UserInfo", user);
+        this.authService.isAuthenticated$.subscribe(isAuthenticated => {
```

Comment:
if you don't unsubscribe subscriptions it may cause memory leaks. Cause subscriptions are not cleaning by garbage collector.